zerver/views/development/registration.py

```diff
@@ -36,7 +36,7 @@ def register_development_user(request: HttpRequest) -> HttpResponse:
                                               Confirmation.USER_REGISTRATION)
     key = activation_url.split('/')[-1]
     # Need to add test data to POST request as it doesnt originally contain the required parameters
-    modify_postdata(request, key=key, full_name=name, password='test', terms='true')
+    modify_postdata(request, key=key, full_name=name, password='test', terms='true', time_format="24")
```

Comment:
If we're trying to set default to '24-hour' format, why not change the `realm.twenty_four_hour_format` to `True`?

@rishig @timabbott ^

Reply:
actually we are not trying to set the default to 24 hrs we just need to check the user browser locale and make changes according to that @vrongmeal